### PR TITLE
sort by full path name instead of by GUID when updating cachefile

### LINKED ISSUE #8108: zpool device name mixup (wwn-....  / scsi-....)
### System information

Type | Version/Name
 --- | --- 
Distribution Name	|  CentOS
Distribution Version	|  7.5.1804
Linux Kernel	|  3.10.0-862.11.6.el7.x86_64
Architecture	| x64
ZFS Version	| 0.7.11-1
SPL Version	| 0.7.11-1

### Describe the problem you're observing
I have labeld my hdd trays with WWN id for proper identification and tried to make zpool status output consistent to reflect naming scheme by wwn - but for my curiousity one of the disk reproducably shows up with scsi-.... name instead of wwn-.... name for no apparent reason.

backupvm1 is a virtual machine with 6 disks attached to a jbod raid controller with VMDirectPath IO passtrough to the vm - 4 disks are raidz1/zfspool, 2 disks are md-raid.  

but imho it should make no difference to physical setup - as you can see the symlinks in /dev/by-id point to the same sdX disk name. 

looks like a zfs/scanning issue to me

```
[root@backupvm1 ~]# zpool status -v
  pool: zfspool
 state: ONLINE
  scan: scrub repaired 0B in 22h11m with 0 errors on Sat Sep 22 15:53:37 2018
config:

	NAME                              STATE     READ WRITE CKSUM
	zfspool                           ONLINE       0     0     0
	  raidz1-0                        ONLINE       0     0     0
	    wwn-0x5000cca260e2f247-part1  ONLINE       0     0     0
	    wwn-0x5000cca260f190fb-part1  ONLINE       0     0     0
	    wwn-0x5000cca260f18f69-part1  ONLINE       0     0     0
	    scsi-35000cca260f18ff7-part1  ONLINE       0     0     0

errors: No known data errors


[root@backupvm1 ~]# ls -la /dev/disk/by-id/
insgesamt 0
drwxr-xr-x 2 root root 680  8. Nov 2018  .
drwxr-xr-x 8 root root 160  8. Nov 2018  ..
lrwxrwxrwx 1 root root   9  8. Nov 2018  ata-VMware_Virtual_SATA_CDRW_Drive_00000000000000000001 -> ../../sr0
lrwxrwxrwx 1 root root  10  8. Nov 2018  dm-name-cl-root -> ../../dm-0
lrwxrwxrwx 1 root root  10  8. Nov 2018  dm-name-cl-swap -> ../../dm-1
lrwxrwxrwx 1 root root  10  8. Nov 2018  dm-uuid-LVM-joL02krkDVA7sxa4q9H11ZxQpEfOKyTDF1I0OFcr4xlr1Zpu5AaEh9JI4ishBfwm -> ../../dm-0
lrwxrwxrwx 1 root root  10  8. Nov 2018  dm-uuid-LVM-joL02krkDVA7sxa4q9H11ZxQpEfOKyTDNIKE6qpMHRmBYlx2TYe0NegSvk75WIpo -> ../../dm-1
lrwxrwxrwx 1 root root  10  8. Nov 2018  lvm-pv-uuid-h9wdKV-a1rf-fnm5-wsOB-Y3SD-nyVR-Q2TY9W -> ../../sda2
lrwxrwxrwx 1 root root  11  8. Nov 2018  md-name-xo-backup-raid1 -> ../../md127
lrwxrwxrwx 1 root root  11  8. Nov 2018  md-uuid-2784a755:54492a1f:e4a3abb3:a707b54b -> ../../md127
lrwxrwxrwx 1 root root   9  8. Nov 2018  scsi-35000cca260e2f247 -> ../../sdc
lrwxrwxrwx 1 root root  10  8. Nov 2018  scsi-35000cca260e2f247-part1 -> ../../sdc1
lrwxrwxrwx 1 root root   9  8. Nov 2018  scsi-35000cca260f18f69 -> ../../sdf
lrwxrwxrwx 1 root root  10  8. Nov 2018  scsi-35000cca260f18f69-part1 -> ../../sdf1
lrwxrwxrwx 1 root root   9  8. Nov 2018  scsi-35000cca260f18ff7 -> ../../sde
lrwxrwxrwx 1 root root  10  8. Nov 2018  scsi-35000cca260f18ff7-part1 -> ../../sde1
lrwxrwxrwx 1 root root   9  8. Nov 2018  scsi-35000cca260f190fb -> ../../sdb
lrwxrwxrwx 1 root root  10  8. Nov 2018  scsi-35000cca260f190fb-part1 -> ../../sdb1
lrwxrwxrwx 1 root root   9  8. Nov 2018  scsi-35000cca261c58b31 -> ../../sdg
lrwxrwxrwx 1 root root  10  8. Nov 2018  scsi-35000cca261c58b31-part1 -> ../../sdg1
lrwxrwxrwx 1 root root   9  8. Nov 2018  scsi-35000cca261c59c9f -> ../../sdd
lrwxrwxrwx 1 root root  10  8. Nov 2018  scsi-35000cca261c59c9f-part1 -> ../../sdd1
lrwxrwxrwx 1 root root   9  8. Nov 2018  wwn-0x5000cca260e2f247 -> ../../sdc
lrwxrwxrwx 1 root root  10  8. Nov 2018  wwn-0x5000cca260e2f247-part1 -> ../../sdc1
lrwxrwxrwx 1 root root   9  8. Nov 2018  wwn-0x5000cca260f18f69 -> ../../sdf
lrwxrwxrwx 1 root root  10  8. Nov 2018  wwn-0x5000cca260f18f69-part1 -> ../../sdf1
lrwxrwxrwx 1 root root   9  8. Nov 2018  wwn-0x5000cca260f18ff7 -> ../../sde
lrwxrwxrwx 1 root root  10  8. Nov 2018  wwn-0x5000cca260f18ff7-part1 -> ../../sde1
lrwxrwxrwx 1 root root   9  8. Nov 2018  wwn-0x5000cca260f190fb -> ../../sdb
lrwxrwxrwx 1 root root  10  8. Nov 2018  wwn-0x5000cca260f190fb-part1 -> ../../sdb1
lrwxrwxrwx 1 root root   9  8. Nov 2018  wwn-0x5000cca261c58b31 -> ../../sdg
lrwxrwxrwx 1 root root  10  8. Nov 2018  wwn-0x5000cca261c58b31-part1 -> ../../sdg1
lrwxrwxrwx 1 root root   9  8. Nov 2018  wwn-0x5000cca261c59c9f -> ../../sdd
lrwxrwxrwx 1 root root  10  8. Nov 2018  wwn-0x5000cca261c59c9f-part1 -> ../../sdd1

[root@backupvm1 ~]# cat /proc/mdstat
Personalities : [raid1]
md127 : active raid1 sdg1[1] sdd1[0]
      7813894464 blocks super 1.2 [2/2] [UU]
      bitmap: 0/59 pages [0KB], 65536KB chunk

unused devices: <none>
```

### Describe how to reproduce the problem
zpool export zfspool
zpool import -d /dev/disk/by-id zfspool

### PR DESCRIPTION
**Pull request by kpande**
closes #8108

Authored-by: Brian Behlendorf <behlendorf1@llnl.gov>
Signed-off-by: Kash Pande <kash@tripleback.net>

### How Has This Been Tested?
Built against git, imported pool using cachefile.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: lib/libzutil/zutil_import.c [hunk 1]
```diff
@@ -1475,11 +1475,11 @@ slice_cache_compare(const void *arg1, const void *arg2)
 	uint64_t guid2 = ((rdsk_node_t *)arg2)->rn_vdev_guid;
 	int rv;
 
-	rv = AVL_CMP(guid1, guid2);
+	rv = AVL_ISIGN(strcmp(nm1, nm2));
```
Comment (richardelling) [inline lib/libzutil/zutil_import.c]: doesn't this change conflict with the comment block above?

Comment (kpande) [reply lib/libzutil/zutil_import.c]: no, it still adds both.

Comment (richardelling) [reply lib/libzutil/zutil_import.c]: nit: should read "Sorted by full path then vdev guid to allow..."

**Comment by kpande** [comment]
updated comment.

**Pull request merged**